src/ddmd/func.d

```diff
@@ -739,6 +739,10 @@ extern (C++) class FuncDeclaration : Declaration
 
             default:
                 {
+                    // https://issues.dlang.org/show_bug.cgi?id=16273
+                    if (!cd.vtbl.dim)
+                        return;
```

Comment:
The trouble with the abrupt `return` is it leaves the FuncDeclaration's semantic result in an indeterminate state, because it didn't finish and no error was generated. A better approach is to attempt to do the semantic analysis on `cd`, and if there is still no `vtbl[]`, then issue an error about a forward reference to `cd`.

Reply:
I tried calling `cd.semantic(null)` here, but that doesn't work because it's already in `cd.semantic`.